dtests/src/test/java/io/snappydata/hydra/cluster/SnappyTest.java

```diff
@@ -3214,62 +3213,180 @@ protected void killVM(String vmDir, String clientName, String vmName) {
+      pb.redirectOutput(ProcessBuilder.Redirect.appendTo(thrDumpOutput));
+      pr = pb.start();
+      pr.waitFor();
+    } catch (IOException ie) {
+      throw new TestException("IOException occurred while retriving logFile path.\nError " +
+          "Message:" + ie.getMessage());
+    } catch (InterruptedException ie) {
+      String s = "Exception occurred while waiting for the process execution : " + pr;
+      throw new TestException(s, ie);
+    }
+  }
+
+  protected void evaluateThrdTypeCounts(Set<String> pidList) {
+    String thrDumpScript = getCurrentDirPath() + File.separator + "jstackDump.sh";
+    Process pr = null;
+    ProcessBuilder pb;
+    File logFile = new File(thrDumpScript);
+    try {
+      if (!logFile.exists()) {
+        FileWriter fw = new FileWriter(logFile.getAbsoluteFile(), true);
+        BufferedWriter bw = new BufferedWriter(fw);
+        for (String pidString : pidList) {
+          int pid = Integer.parseInt(pidString);
+          String host = snappyTest.getPidHost(Integer.toString(pid));
+          if (host.equalsIgnoreCase("localhost")) {
+            bw.write("jstack " + pid + " > " + pid + "_dump.txt");
+          } else {
+            bw.write("ssh -n -x -o PasswordAuthentication=no -o StrictHostKeyChecking=no " +
+                host + " jstack " + pid + " > " + pid + "_dump.txt");
+          }
           bw.newLine();
-          bw.write("ssh -n -x -o PasswordAuthentication=no -o StrictHostKeyChecking=no " +
-              pidHost + " kill -3 " + pid);
         }
-        bw.newLine();
+        bw.close();
+        fw.close();
+        logFile.setExecutable(true);
       }
-      bw.close();
-      fw.close();
-      logFile.setExecutable(true);
-      pb = new ProcessBuilder(server);
+      String thrDumpLog = getCurrentDirPath() + File.separator + "jStackDump.log";
+      File thrDumpOutput = new File(thrDumpLog);
+      pb = new ProcessBuilder(thrDumpScript);
       pb.redirectErrorStream(true);
-      pb.redirectOutput(ProcessBuilder.Redirect.appendTo(serverHeapDumpOutput));
+      pb.redirectOutput(ProcessBuilder.Redirect.appendTo(thrDumpOutput));
       pr = pb.start();
       pr.waitFor();
-    } catch (IOException e) {
-      throw new TestException("IOException occurred while retriving logFile path " + log + "\nError Message:" + e.getMessage());
-    } catch (InterruptedException e) {
+
+      for (String pidString : pidList) {
+        Map<String,Integer> thrMap = new HashMap<String, Integer>();
+        try {
+          String fileName = getCurrentDirPath() + File.separator + pidString + "_dump.txt";
+          String line = null;
+          String map_id = null;
+          BufferedReader br = new BufferedReader(new FileReader(fileName));
+          int totalThrdCnt = 0;
+          while ((line = br.readLine()) != null) {
+            if (line.startsWith("\"")) {
+              String thrdName = line.substring(line.indexOf("\"") + 1, line.lastIndexOf("\""));
```

Comment:
Check for another effective approach(checking against the list of expected thread group names) to find out the thread group name and the total count against it. Thread group names displayed at the end in collective stats should be complete/informative.

Reply:
This can be done in the next PR.